### ISSUE TITLE
[Feature] save-exact and opt out of yarn.lock

### ISSUE THREAD

**Issue by Bessonov** [yarnpkg/berry#299]
**Describe the user story**

I've noticed, that some options are not available. I would like to save exact versions of dependencies in `package.json` automatically. I'm aware, that this is not working for transitive dependencies.

Furthermore I would like to opt out of yarn.lock.

**Comment by arcanis** [comment]
The first one makes sense, but I'm afraid the second one won't happen as it doesn't fit the purpose of the tool.

**Comment by Bessonov** [comment]
It's OK that second one wouldn't be supported. It's a bug or feature, that absolute paths are not supported?

```
yarn config set lockfileFilename /tmp/random-yarn.lock

[...]

YN0001: Error: ENOENT: no such file or directory, open '/home/vagrant/workspace/tmp/random-yarn.lock'
```
?

**Comment by arcanis** [comment]
The `*Filename` settings are expected to be filenames, not paths (it matters because otherwise we would have to add special logic to support them when looping over multiple levels of filesystem hierarchy). We technically should prevent them from being absolute paths.

**Comment by Bessonov** [comment]
Well, `yarn config set lockfileFilename ../../../../../../../tmp/poc-yarn.lock` seems to work for me :man_dancing:

**Comment by arcanis** [comment]
Fixed in #561
